packages/contracts/src/EmailAuthSigner.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import {EmailProof} from "./utils/Verifier.sol";
+import {IDKIMRegistry} from "@zk-email/contracts/DKIMRegistry.sol";
+import {IVerifier, EmailProof} from "./interfaces/IVerifier.sol";
+import {CommandUtils} from "./libraries/CommandUtils.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {IEmailAuth, EmailAuthMsg} from "./interfaces/IEmailAuth.sol";
+import {ERC1271SignatureValidator} from "./ERC1271SignatureValidator.sol";
+
+/// @title Email Authentication/Authorization Contract for Signature-like Usage
+/// @notice This contract provides a signature-like authentication mechanism using emails.
+/// Similar to how ECDSA signatures work, this contract only verifies the authenticity
+/// of an email command without handling replay protection or nullifiers - these should
+/// be implemented at the application level.
+/// @dev Unlike EmailAuth.sol which handles nullifiers internally, this contract is designed
+/// to be used like a signature verification mechanism where the calling contract manages
+/// its own replay protection.
+contract EmailAuthSigner is
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    IEmailAuth,
+    ERC1271SignatureValidator
+{
+    /// The CREATE2 salt of this contract defined as a hash of an email address and an account code.
+    bytes32 public accountSalt;
+    /// An instance of the DKIM registry contract.
+    address public dkimRegistryAddr;
+    /// An instance of the Verifier contract.
+    address public verifierAddr;
+    /// The templateId of the sign hash command.
+    uint256 public templateId;
+
+    constructor() {}
+
+    /// @notice Initialize the contract with an initial owner, account salt, DKIM registry address, and verifier address.
+    /// @param _initialOwner The address of the initial owner.
+    /// @param _accountSalt The account salt to derive CREATE2 address of this contract.
+    /// @param _dkimRegistryAddr The address of the DKIM registry contract.
+    /// @param _verifierAddr The address of the verifier contract.
+    /// @param _templateId The templateId of the sign hash command.
+    function initialize(
+        address _initialOwner,
+        bytes32 _accountSalt,
+        address _dkimRegistryAddr,
+        address _verifierAddr,
+        uint256 _templateId
+    ) public initializer {
+        __Ownable_init(_initialOwner);
+        accountSalt = _accountSalt;
+        templateId = _templateId;
+        require(
+            _dkimRegistryAddr != address(0),
+            "invalid dkim registry address"
```

Comment:
As discussed in breakout call, lets move to use custom errors if you're in agreement